graphbolt/src/shared_memory_utils.cc

```diff
@@ -18,164 +18,193 @@
 namespace graphbolt {
 namespace sampling {
 
-static SharedMemoryPtr CopyTorchArchiveToSharedMemory(
-    const std::string& name, int64_t size,
-    torch::serialize::OutputArchive& archive) {
-  std::stringstream serialized;
-  archive.save_to(serialized);
-  auto serialized_str = serialized.str();
-  auto shm = std::make_unique<SharedMemory>(name);
-  auto mem_buf = shm->Create(size);
-  // Use the first 8 bytes to store the size of the serialized string.
-  static_cast<int64_t*>(mem_buf)[0] = serialized_str.size();
-  memcpy(
-      (char*)mem_buf + sizeof(int64_t), serialized_str.data(),
-      serialized_str.size());
-  return shm;
+static std::string GetSharedMemoryMetaName(const std::string& name) {
+  return name + "_meta";
 }
 
-static SharedMemoryPtr LoadTorchArchiveFromSharedMemory(
-    const std::string& name, int64_t max_meta_size,
-    torch::serialize::InputArchive& archive) {
-  auto shm = std::make_unique<SharedMemory>(name);
-  auto mem_buf = shm->Open(max_meta_size);
-  int64_t meta_size = static_cast<int64_t*>(mem_buf)[0];
-  archive.load_from(
-      static_cast<const char*>(mem_buf) + sizeof(int64_t), meta_size);
-  return shm;
+static std::string GetSharedMemoryDataName(const std::string& name) {
+  return name + "_data";
 }
 
-static SharedMemoryPtr CopyTensorsDataToSharedMemory(
-    const std::string& name,
-    const std::vector<torch::optional<torch::Tensor>>& tensors) {
-  int64_t memory_size = 0;
-  for (const auto& optional_tensor : tensors) {
-    if (optional_tensor.has_value()) {
-      auto tensor = optional_tensor.value();
-      memory_size += tensor.numel() * tensor.element_size();
-    }
-  }
-  auto shm = std::make_unique<SharedMemory>(name);
-  auto mem_buf = shm->Create(memory_size);
-  for (auto optional_tensor : tensors) {
-    if (optional_tensor.has_value()) {
-      auto tensor = optional_tensor.value().contiguous();
-      int64_t size = tensor.numel() * tensor.element_size();
-      memcpy(mem_buf, tensor.data_ptr(), size);
-      mem_buf = static_cast<char*>(mem_buf) + size;
-    }
+// To avoid unaligned memory access, we round the size of the binary buffer to
+// the nearest multiple of 256 bytes.
+inline static int64_t GetRoundedSize(int64_t size) {
+  constexpr int64_t ALIGNED_SIZE = 8;
+  return (size + ALIGNED_SIZE - 1) / ALIGNED_SIZE * ALIGNED_SIZE;
+}
+
+SharedMemoryHelper::SharedMemoryHelper(
+    const std::string& name, int64_t max_meta_memory_size, bool is_creator)
```

Comment:
```suggestion
    const std::string& name, bool readonly, int64_t max_metadata_size)
```

Reply:
`is_creator` is not a good name. But I think the name `readonly` might also be inaccurate, which does not carry the message that the writer will creates a new shared memory object.